video_loader.py

```diff
@@ -0,0 +1,134 @@
+__copyright__ = "Copyright (c) 2020-2021 Jina AI Limited. All rights reserved."
+__license__ = "Apache-2.0"
+
+import io
+import os
+import random
+import string
+import tempfile
+import urllib.request
+
+import ffmpeg
+import librosa
+import numpy as np
+from jina import Document, DocumentArray, Executor, requests
+from jina.logging.logger import JinaLogger
+from jina.types.document import _is_datauri
+
+DEFAULT_FPS = 1
+
+
+class VideoLoader(Executor):
+    """
+    An executor to extract the image frames, audio, and the subtitle from videos with `ffmpeg`
+    """
+
+    def __init__(self, max_num_frames: int = 50, fps=DEFAULT_FPS, **kwargs):
+        """
+        :param max_num_frames: maximum number of images frames to be extracted from the video
+        :param fps: number of frames extracted per second
+        :param kwargs: the **kwargs for Executor
+        """
+        super().__init__(**kwargs)
+        self.max_num_frames = max_num_frames
+        self.fps = fps
+        self.logger = JinaLogger(
+            getattr(self.metas, 'name', self.__class__.__name__)
+        ).logger
+
+    @requests
+    def extract(self, docs: DocumentArray, **kwargs):
+        """
+        Load the video from the Document.uri, extract frames, audio, subtitle
+        and save it into chunks.
+
+        :param docs: the input Documents with either the video file name or URL in the `uri` field
+        """
+        for doc in docs:
+            self.logger.info(f'received {doc.id}')
+
+            if doc.uri == '':
+                raise ValueError(f'No uri passed for the Document: {doc.id}')
+
+            with tempfile.TemporaryDirectory() as tmpdir:
+                source_fn = (
+                    self._save_uri_to_tmp_file(doc.uri, tmpdir)
+                    if _is_datauri(doc.uri)
+                    else doc.uri
+                )
+
+                # extract all the frames video
+                frame_fn_list = self._convert_video_uri_to_frames(source_fn)
+
+                # add frames as chunks to the Document, with modality='image'
+                for idx, frame_fn in enumerate(frame_fn_list):
+                    self.logger.debug(f'frame: {idx}')
+                    _chunk = Document(modality='image')
```

Comment:
```suggestion
                    chunk = Document(modality='image')
```
No point prefixing variable names with `_` if they are accessible only inside function scope